src/Stratis.Bitcoin.Features.Consensus/Rules/ConsensusRules.cs

```diff
@@ -181,4 +187,36 @@ public void SetErrorHandler(ConsensusRule errorHandler)
             }
         }
     }
+
+    public static class ConsensusRulesExtension
+    {
+        /// <summary>
+        /// Try to find a consensus rule in the rules collection.
+        /// </summary>
+        /// <typeparam name="T">The type of rule to find.</typeparam>
+        /// <param name="rules">The rules to look in.</param>
+        /// <returns>The rule or <c>null</c> if not found in the list.</returns>
+        public static T TryFindRule<T>(this IEnumerable<ConsensusRule> rules) where T : ConsensusRule
```

Comment:
With TryXXX methods we're used to an out parameter containing the result value if successful and a bool return type.
Maybe this method could be changed to something reflecting this convention.

Reply:
That's not always correct the `TryXXX` name convention means this will not throw and not always has an out, but rather may return a null so the caller knows this method is will not fail if the entity is not found.